archive_viewer/widgets/item_delegates.py

```diff
@@ -5,11 +5,42 @@
 from qtpy.QtWidgets import (QStyledItemDelegate, QSlider, QComboBox, QStyle,
                             QPushButton, QTableView, QStyleOptionViewItem,
                             QWidget, QDoubleSpinBox, QLineEdit)
-from config import logger
 from widgets import (ColorButton, CenterCheckbox)
 
 
-class SliderDelegate(QStyledItemDelegate):
+class EditorDelegate(QStyledItemDelegate):
+    def __init__(self, parent: QTableView) -> None:
+        super().__init__(parent)
+        self.editor_list = []
+        model = self.parent().model()
+        model.modelAboutToBeReset.connect(self.reset_editors)
+
+    def paint(self, painter: QPainter, option: QStyleOptionViewItem, index: QModelIndex) -> None:
+        """Create a new persistent editor on the Table View at the given index."""
```

Comment:
can you add the return and parameters to the doc for the methods added by this PR? Thanks

Reply:
I updated the docstrings and return parameters